client/packages/common/src/ui/layout/tables/components/Cells/MultiNumberInputCell/MultiNumberInputCell.tsx

```diff
@@ -0,0 +1,167 @@
+  key: string;
+  ratio: number;
+  label: string;
+}
+
+export const MultipleNumberInputCell = <T extends RecordWithId>({
+  units,
+  rowData,
+  column,
+  rowIndex,
+  columnIndex,
+  isDisabled = false,
+  min = 0,
+  max,
+  decimalLimit,
+  step,
+  multiplier,
+  defaultValue,
+  allowNegative,
+  id,
+  TextInputProps,
+  width,
+}: CellProps<T> &
+  NumericInputProps & {
+    id?: string;
+    TextInputProps?: StandardTextFieldProps;
+  } & { units: Unit[] }): React.ReactElement<CellProps<T>> => {
+  const [buffer, setBuffer] = useBufferState(column.accessor({ rowData }));
+  const updater = useDebounceCallback(column.setter, [column.setter], 350);
+
+  const autoFocus = rowIndex === 0 && columnIndex === 0;
+
+  const cellValues = getCellValues(buffer as number | undefined, units);
+
+  return (
+    <Box display="flex">
+      {units.map((unit, index) => (
+        <NumericTextInput
+          key={unit.key}
+          id={id}
+          disabled={isDisabled}
+          autoFocus={autoFocus}
+          {...TextInputProps}
+          InputProps={{
+            sx: { '& .MuiInput-input': { textAlign: 'right' } },
+            ...TextInputProps?.InputProps,
+          }}
+          onChange={num => {
+            const newValue = num === undefined ? min : num;
+            if (cellValues[index] === newValue) return;
+            const newValues = [...cellValues];
+            newValues[index] = newValue;
+            const newTotal = computeTotal(newValues, units);
+            setBuffer(newTotal);
+            updater({ ...rowData, [column.key]: Number(newTotal) });
+          }}
+          min={min}
+          // Limit smaller units to less than the value of the next highest
+          // unit, e.g. "Months" would be limited to 11.9999
+          max={setMax(units, index, max)}
+          decimalLimit={decimalLimit}
+          step={step}
+          multiplier={multiplier}
+          allowNegative={allowNegative}
+          defaultValue={defaultValue}
+          value={cellValues[index]}
+          width={width}
+          endAdornment={unit.label}
+        />
+      ))}
+    </Box>
+  );
+};
+
+export const getCellValues = (value: number | undefined, units: Unit[]) => {
+  if (value === undefined) return units.map(_ => 0);
+
+  let remainder = value;
+  return units.map((unit, index) => {
+    const thisValue =
+      index === units.length - 1
+        ? remainder
+        : Math.floor(remainder / unit.ratio);
+    // Rounding required to binary precision problems
+    // e.g. 66.6 % 12 => 6.599999999999994
+    remainder = parseFloat((remainder % unit.ratio).toFixed(5));
+    return thisValue;
+  });
+};
+
+export const computeTotal = (values: number[], units: Unit[]) =>
+  values.reduce((sum, val, index) => {
+    const ratio = units[index]?.ratio ?? 1;
+    return sum + ratio * val;
+  }, 0);
+
+export const setMax = (
+  units: Unit[],
+  index: number,
+  max: number | undefined
+) => {
+  if (index === 0) return max;
```

Comment:
I made the component's `max` property applicable to whatever the highest unit is (i.e. years). But now that I think about it, it should probably be in terms of the *base* unit, right? Makes the logic a tad more complex, but probably worth it? (It would also be more consistent with the `min` prop)

Reply:
Hm, max years actually makes more sense to me than max months, though I see your point... I think document as a comment, and see if there is any feedback if/when this component gets reused 🙏